source/_posts/2020-09-12-android-240-release.markdown

```diff
@@ -0,0 +1,67 @@
+---
+title: "Home Assistant Companion Android App Release 2.4.0"
+description: "What's new with the Home Assistant Companion Android App in 2.4.0"
+date: 2020-09-12 00:00:00
+date_formatted: "September 12, 2020"
+comments: true
+author: Daniel Shokouhi
+categories: Release-Notes
+og_image: /images/blog/2020-09-12-android-240-release/Companion.png
+---
+
+Hey there, hope everyone has been enjoying the last release of the Android app. We have another update coming up where we felt its time to give everyone a heads up before release. There are several new changes done that you will immediately notice after installing the update to 2.4.0 so we wanted to cover whats new. You can expect this release to be ready in the next day in the Google Play Store.
+
+## Current and Future Breaking Changes
+
+Sometimes when we move really fast we may move a bit too fast. While we had fun adding all the new sensors we didn't stop to think about the new user experience and how we are making a burden on them by having non-static attributes for some of the sensors. We wanted to fix this issue but also be curteous to our current users so we have made a few changes. Some of which take place in this release and some in the following release after 2.4.0.
+
+The first major change done that takes place in this release is that the app will no longer send needless updates to your Home Assistant server when there is nothing new to report state-wise. This is the driving change as to why you will see more sensors in the Manage Sensor screen. Some of the attributes attached to some sensors update outside of the state so to combat this change we have split those attributes to become their own sensor. We have opted to keep the existing attributes where they are for this release so they do not break existing automations, they may trigger slower until you move to the new sensor. All of the new sensors are also disabled by default (with the exception of battery) so you will want to enable the ones you care for. The following sensors have had their attributes split up:
+
+* Audio Sensor - All attributes. The sensor name for new users has also been changed to `Ringer Mode` to better align with what the state represents.
+* Battery State Sensor - All attributes.
+* Bluetooth Connection - Only `is_bt_on` has been made into its own binary sensor. The other attributes update with the state.
+* Storage Sensor - External storage attributes. The new sensor will behave the same as the current storage sensor which has also now been labeled as Internal Storage.
+* WiFi Connection - All attributes with the exception of `is_hidden`.
+
+These attributes mentioned above will be removed in the next release so please make your switch to the new sensors now to avoid the breaking change then.
```

Comment:
```suggestion
These attributes mentioned above will be removed in the next release. Please make your switch to the new sensors now to avoid the breaking change then.
```